src/backend/strategies.rs

```diff
@@ -922,116 +931,159 @@ pub(crate) async fn run_strategy_task<'s>(
-                                                        // Log the detailed state transition broadcast error
-                                                        error!("State Transition Broadcast Error: Code: {}, Message: \"{}\"", error.code, error.message);
-                                                    }
-                                                }
-
-                                                // If a data contract was registered, add it to
-                                                // known_contracts
-                                                if let StateTransition::DataContractCreate(
-                                                    DataContractCreateTransition::V0(
-                                                        data_contract_create_transition,
-                                                    ),
-                                                ) = &transition
-                                                {
-                                                    // Extract the data contract from the transition
-                                                    let data_contract_serialized =
-                                                        &data_contract_create_transition
-                                                            .data_contract;
-                                                    let data_contract_result =
-                                                        DataContract::try_from_platform_versioned(
-                                                            data_contract_serialized.clone(),
-                                                            false,
-                                                            PlatformVersion::latest(),
+                                                Some(if let Some(wait_for_state_transition_result_response::Version::V0(v0_response)) = &wait_response.version {
+                                                    if let Some(metadata) = &v0_response.metadata {
+                                                        let actual_block_height = metadata.height;
+                                                        info!(
+                                                            "Successfully processed state transition {} ({}) for block {} (Actual block height: {})",
+                                                            index + 1, transition.name(), current_block_info.height, actual_block_height
                                                         );
+                                    
+                                                        // Verification of the proof
+                                                        if let Some(wait_for_state_transition_result_response_v0::Result::Proof(proof)) = &v0_response.result {
+                                                            // For proof verification, if it's a DocumentsBatch, include the data contract, else don't
```

Comment:
Don't you want both if there is a proof and if there isn't?

Reply:
Basically so we can test both out.

Reply:
There's no functionality for that in the sdk right now so we'll have to add it there first.